pkg/containerd/events.go

```diff
@@ -222,8 +236,13 @@ func (e *EventMonitor) initializeExistingContainers(ctx context.Context) {
 	}
 }
 
-func (e *EventMonitor) createPortMapping(ctx context.Context, containerID string) (nat.PortMap, error) {
-	container, err := e.containerdClient.ContainerService().Get(ctx, containerID)
+func (e *EventMonitor) createPortMapping(ctx context.Context, ns, containerID string) (nat.PortMap, error) {
+	client, err := containerd.New(e.containerdSocket, containerd.WithDefaultNamespace(ns))
```

Comment:
The `WithNamespace` will not work since it only returns a context that could only be used with a specific client. 
I also attempted the `client.Containers(filter: string[]{fmt.Sprintf("namespace=%s", ns)})` that did not really work since the containers are all the containers in the same namespace that client-initiated with. 

I think I will try re-using the GRPC connection that looks promising.